arangod/Cluster/ClusterTypes.h

```diff
@@ -42,6 +42,15 @@ typedef std::string ViewID;           // ID of a view
 typedef std::string ShardID;          // ID of a shard
 typedef uint32_t ServerShortID;       // Short ID of a server
 typedef std::string ServerShortName;  // Short name of a server
+namespace pmr {
+typedef std::pmr::string ServerID;         // ID of a server
```

Comment:
ServerID is fixed size string (PRMR/COORD/...-uuid), I think better to use array<char, ...>

Reply:
out of scope.

Reply:
I don't think so, you already changed the type, why don't change to simpler and smaller one?

Reply:
I think so. ServerIDs are not fixed size. In fact the server name can be dictated from the outside, as it is done for kubernetes.

Reply:
> ServerIDs are not fixed size.

Really? :(

> In fact the server name can be dictated from the outside, as it is done for kubernetes.

Could you explain for me where ServerID set to not restricted size string?
I found only two places and both looks like role + uuid

```cpp
  std::string id;
  bool hadPersistedId = hasPersistedId();
  if (!hadPersistedId) {
    id = generatePersistedId(role);

    LOG_TOPIC("0d924", INFO, Logger::CLUSTER)
        << "Fresh start. Persisting new UUID " << id;
  } else {
    id = getPersistedId();
    LOG_TOPIC("db3ce", DEBUG, Logger::CLUSTER)
        << "Restarting with persisted UUID " << id;
  }
  setId(id);
```

```cpp
    // If we have persisted id, we use that. Otherwise we check, if we were
    // given a disaster recovery id that wins then before a new one is generated
    // and that choice persisted.
    std::string uuid;
    if (ServerState::instance()->hasPersistedId()) {
      uuid = ServerState::instance()->getPersistedId();
    } else {
      std::string recoveryId = _agent->config().recoveryId();
      if (recoveryId.empty()) {
        uuid = ServerState::instance()->generatePersistedId(
            ServerState::ROLE_AGENT);
      } else {
        uuid = recoveryId;
        ServerState::instance()->writePersistedId(recoveryId);
      }
    }
    _agent->id(uuid);
    ServerState::instance()->setId(uuid);
```


Maybe this option?
```cpp
  options->addOption("--agency.disaster-recovery-id",
                     "Specify the ID for this agent. WARNING: This is a "
                     "dangerous option, for disaster recover only!"
```

Reply:
> Or we just put something not uuid like to the uuid file?
Yes.

Reply:
Ok